python-package/lightgbm/dask.py

```diff
@@ -68,15 +68,22 @@ def _train_part(params, model_factory, list_of_parts, worker_addresses, return_m
     network_params = _build_network_params(worker_addresses, get_worker().address, local_listen_port, time_out)
     params.update(network_params)
 
+    is_ranker = model_factory.__qualname__ == 'LGBMRanker'
+
     # Concatenate many parts into one
-    parts = tuple(zip(*list_of_parts))
-    data = _concat(parts[0])
-    label = _concat(parts[1])
-    weight = _concat(parts[2]) if len(parts) == 3 else None
+    data = _concat([d['X'] for d in list_of_parts])
+    label = _concat([d['y'] for d in list_of_parts])
+    weight = _concat([d['weight'] for d in list_of_parts]) if 'weight' in list_of_parts[0] else None
```

Comment:
I love this change, thank you so much!

Since `list_of_parts` is already guaranteed to all be data sitting on this worker at this point, I don't think we were getting any value from the `tuple()` thing.

Can you make this a separate PR? I think we could review and merge that pretty quickly, and that it's not directly related to the ranker changes.

Reply:
Sure, sorry this is my first OS PR and I guess the scope of the changes here is more than just "add ranker support" and is a non-insignificant stray from how xgboost.dask is written. Will do.

Reply:
no prob! I have a preference for smaller PRs that we can review quickly, since I'm also actively working on the Dask module right now. Reduces the risk that we cause merge conflicts for each other, yenno

Reply:
Addressed in d6e5209